include/rovio/PoseUpdate.hpp

```diff
@@ -60,8 +60,10 @@ class PoseUpdateMeas: public LWF::State<LWF::VectorElement<3>,LWF::QuaternionEle
  public:
   static constexpr unsigned int _pos = 0;
   static constexpr unsigned int _att = _pos+1;
+  M3D pos_cov_; // Will be used to scale the update covariance according to the measurement
```

Comment:
I'd add another flag for the other 3x3 block though, do you agree? For
example, my gps drivers sets the orientation diagonal to -1, so I wouldn't
want to use that..

On Friday, 5 August 2016, bloesch <notifications@github.com> wrote:

> In include/rovio/PoseUpdate.hpp
> <https://github.com/ethz-asl/rovio/pull/78#discussion_r73650365>:
>
> > @@ -60,8 +60,10 @@ class PoseUpdateMeas: public LWF::State<LWF::VectorElement<3>,LWF::QuaternionEle
> >   public:
> >    static constexpr unsigned int _pos = 0;
> >    static constexpr unsigned int _att = _pos+1;
> > +  M3D pos_cov_; // Will be used to scale the update covariance according to the measurement
>
> I think it would be better to use the full 6x6 matrix. More generic and
> still easy to use.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ethz-asl/rovio/pull/78/files/f472b3bc6a3173bd9d195bfb42d4197e965d1eea#r73650365>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AKO4Byvwc5uC9CDAYpOwX3Fs3vr792vCks5qcuEJgaJpZM4JcIxM>
> .
>